base/compiler/tfuncs.jl

```diff
@@ -1817,6 +1855,10 @@ function builtin_effects(f::Builtin, argtypes::Vector{Any}, @nospecialize(rt))
         end
         s = s::DataType
         consistent = !ismutabletype(s) ? ALWAYS_TRUE : ALWAYS_FALSE
+        # access to undefined field leads to undefined behavior and should taint `:consistent`-cy
```

Comment:
Make a note that undefined access of non-isbitstype would be ok, but we model it conservatively for simplicitly?